### ISSUE TITLE
Python version in toolchain docs

### ISSUE THREAD

**Issue by taylorstapleton** [googlevr/seurat#5]
It seems that for building ion:

`c:\Seurat\third\_party\\ion>ion\build.bat -c opt_x64`

a specific version of Python is required to be installed (2.7 I believe). This might be worth mentioning in the supported-toolchain documentation.

**Comment by mebalzer** [comment]
Yes you are correct, as well as the Windows command line entries are wrong, e.g. "third/_party" instead of "third_party."

Also, the python is looking for a gyp file.

It seems this was not actually tested on a Windows system, did you get it to build?

**Comment by taylorstapleton** [comment]
No, not yet.

Interestingly, I didn't even notice the third/_party thing was a mistake until now. I have been working out of that subdir structure the whole time. :-p

I also had to fix a path in one of the gyp files that was building a path with a back-slash instead of forward-slash. It seems it might be possible to keep playing whack-a-mole on build errors for a while though before getting it.

**Comment by mebalzer** [comment]
Yea I gave up when I realized the Unreal plug-in did not include a binary either, which is the direction I am going since I had specific need for Seurat three months ago.  Sadly, I think this was more for show for Google IO and to check off a box, than actually have a library ready.  This is not the first time I have spent hours having to go through like you are and play "wack-a-mole."  I may spend a couple of hours today again, and report anymore errors I come up with in your thread.  If you succeed and you have an actually have build, please post it, I know there are many who would be very happy.

**Comment by mebalzer** [comment]
Here is the directory structure I think was intended:

Seurat\third_party\ion\>ion\build.bat -c opt_x64  (I had to edit my build.bat to point to my c:\python27 directory).

However, I received this error message: 
ERROR: Could not determine what you want to build. Try specifying a gyp file or directory containing one.

If I 'cd' to the ion and run >build.bat -c opt_x64 , I get this:  e:\seurat\third_party\ion\third_party\icu\icu4c\source\common\unicode\ptypes.h(43): fatal error C1083: Cannot open include file: 'stddef.h': No such file or directory
ninja: build stopped: subcommand failed.
(There are several other errors, but the are pretty much the same "No such file or directory."

**Comment by mebalzer** [comment]
Hats off to Dmitri for plugging through and compiling this in VS2017.  You can find his fork here: https://github.com/ddiakopoulos/seurat/

**Comment by davejrichardson** [comment]
Thanks for reporting your problems!

I do see some problems in the BUILDING.md, perhaps regarding the escaping of backslashes in paths in the Windows instructions. You have to build Ion from, for example, c:\seurat\third_party\ion\ion. Note the two copies of \ion in the path.

We've built Seurat on a Windows 10 machine with Visual Studio 2015 Update 3 and a few Linux+Clang distributions. IIRC, Ion's build batch file assumes it's in the same directory as ion.gyp, and that's probably what it can't find when you build from another folder.

We don't include binaries for Unreal/Unity because there are so many versions of them, and many developers build from source with modifications. Even when we attempted to make plugin binaries for  clean versions of the engine, it was difficult. However, it might be possible to convert our capture tool into a blueprint.

**Comment by davejrichardson** [comment]
Hi @taylorstapleton, would you mind posting the specific error you had with Ion and python 2.7?

I'm working on fixing the documentation bugs.

**Comment by taylorstapleton** [comment]
@davejrichardson 
Thanks for checking in, the specific change I made to move forward in the Ion building process was this one:

> diff --git a/ion/build.py b/ion/build.py
> index 154fe09..56f0afd 100755
> --- a/ion/build.py
> +++ b/ion/build.py
> @@ -77,7 +77,7
> -CONFIGURATION_GYPI = 'dev/os.gypi'
> +CONFIGURATION_GYPI = 'dev\os.gypi'

**Comment by mebalzer** [comment]
Thanks @davejrichardson for checking in and yes it makes sense to not offer an Unreal binary, but I have found even after compiling them in UE4.19.2 that the plug-in does not work, but will add the code snippets in Github for Unreal plugin for further clarification.
